feat: updated version that uses ES6 crypto API

PiiXiieeS opened the pull request:
Here you are a pull request that fixes the concerts of Math.random() not guaranteed to be a high-quality RNG.
Anyway, I understand your remark and the fact that this may not be a serious issue for the use cases managed by this library, like framebus that we have started to use in our projects.

Credits for https://stackoverflow.com/questions/105034/how-to-create-a-guid-uuid

PiiXiieeS committed:
feat: updated version that uses ES6 crypto API

Credits for https://stackoverflow.com/questions/105034/how-to-create-a-guid-uuid

crookedneighbor commented on index.js:
```diff
@@ -1,12 +1,9 @@
 "use strict";
 
 function uuid() {
```
```suggestion
function uuid() {
  // https://stackoverflow.com/a/2117523/2601552
```

crookedneighbor replied:
Let's include the link where you pulled this solution from as a comment in the code.

PiiXiieeS replied:
I found also a problem with the gulp minifying when trying to compile framebus and not handling well the arrow function:
[23:17:29] framebus.min.js 50.1 kB
[23:17:29] 'build' errored after 2.18 s
[23:17:29] GulpUglifyError: unable to minify JavaScript
Caused by: SyntaxError: Unexpected token: operator «>»
File: /Users/jhgascon/Code/framebus/framebus.min.js
Line: 5
Col: 59

Issue is solved with standard function(c) on the other hand. 
Correction submitted.

PiiXiieeS committed:
fix(uuid): remove arrow function

move from arrow function to standard function(c) to avoid issues with minifying with gulp task

PiiXiieeS commented:
fix(uuid): remove arrow function
move from arrow function to standard function(c) to avoid issues with minifying with gulp task

crookedneighbor commented:
This is looking good. My one concern is for browser compatibility. 

[Uint8Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array) is not supported in IE9. 

[Crypto](https://developer.mozilla.org/en-US/docs/Web/API/Crypto) is not supported in IE9 or 10.

There are parts of the Braintree web SDK that still technically support IE9, so we'll need to wait until the next major version bump of the Braintree SDK to incorporate these changes. We're working on that now, so it shouldn't be too long. Until then, I'll leave this PR open.

Thanks so much for working on this! Hoping to get it merged in within the next couple of months.

crookedneighbor commented:
Also, using https://jsbench.me/, it shows it as being 97% slower to run it this way, though for Braintree's use case, I don't think that's a big deal. Running it a few times is still fast enough to not be noticeable by the end user.

@braintree/team-sdk would love to hear y'all's thoughts on this when you get a second.

hacklschorsch commented:
> @braintree/team-sdk would love to hear y'all's thoughts on this when you get a second.

I find it amusing that the SDK of a payment company uses `Math.random()` for randomness :)  Hope nobody of those 340 k downloads per week on NPMJS associates the company name with serious business and uses your library to shoot a hole in their foot 🙏

infame commented:
Closing as resolved — Math.random() usage was removed in v2.0.0.
